extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/MapboxCarMapSurface.kt

```diff
@@ -0,0 +1,33 @@
+package com.mapbox.maps.extension.androidauto
+
+import androidx.car.app.CarContext
+import androidx.car.app.SurfaceContainer
+import com.mapbox.maps.MapSurface
+import com.mapbox.maps.MapboxExperimental
+
+/**
+ * This contains the Android Auto head unit map information.
+ * @see MapboxCarMap.registerObserver
+ *
+ * @property carContext reference to the context provided to the [MapboxCarMap]
+ * @property mapSurface Mapbox controllable interface
+ * @property surfaceContainer A container for the Surface created by the car.
+ */
+@MapboxExperimental
+class MapboxCarMapSurface internal constructor(
+  val carContext: CarContext,
```

Comment:
wouldn't it leak context while keeping a strong reference of CarContext here?

Reply:
If a downstream developer kept a strong reference of the MapboxCarMapSurface, outside of the Car Session lifecycle scope. Yes it would leak. Similar to how, if a developer held a strong reference of `MapInitOptions` with an Activity context.

I was concerned with memory leaks early on, but the `MapboxCarMapObserver` discourages strong references. you handle the surface in `onAttached`. Here is the only memory leak i've seen and it was google internal https://issuetracker.google.com/u/0/issues/203594731